routes/newsletter.js

```diff
@@ -0,0 +1,12 @@
+const express = require("express");
+const router = express.Router();
+const newsletters = require("../controllers/newsletter");
+const { validateEmail } = require("../middlewares/validators/newsletter");
+
+router.post("/subscribe", validateEmail, newsletters.addEmail);
```

Comment:
Is there any reason we are going with `post` request here? Patch request might be better suited as we are updating data as well. 

ref- https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH

Reply:
Post also works the same @pallabez , as internally in the controller piyush is handling for both condition here